bigbluebutton-html5/imports/ui/components/video-preview/component.jsx

```diff
@@ -0,0 +1,231 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { defineMessages, injectIntl, intlShape } from 'react-intl';
+import Button from '/imports/ui/components/button/component';
+import ModalBase from '/imports/ui/components/modal/base/component';
+import { styles } from './styles';
+
+const propTypes = {
+  intl: intlShape.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  startSharing: PropTypes.func.isRequired,
+  changeWebcam: PropTypes.func.isRequired,
+};
+
+const intlMessages = defineMessages({
+  webcamSettingsTitle: {
+    id: 'app.videoPreview.webcamSettingsTitle',
+    description: 'Title for the video preview modal',
+  },
+  closeLabel: {
+    id: 'app.videoPreview.closeLabel',
+    description: 'Close button label',
+  },
+  webcamPreviewLabel: {
+    id: 'app.videoPreview.webcamPreviewLabel',
+    description: 'Webcam preview label',
+  },
+  cameraLabel: {
+    id: 'app.videoPreview.cameraLabel',
+    description: 'Camera dropdown label',
+  },
+  cancelLabel: {
+    id: 'app.videoPreview.cancelLabel',
+    description: 'Cancel button label',
+  },
+  startSharingLabel: {
+    id: 'app.videoPreview.startSharingLabel',
+    description: 'Start Sharing button label',
+  },
+  webcamOptionLabel: {
+    id: 'app.videoPreview.webcamOptionLabel',
+    description: 'Default webcam option label',
+  },
+  webcamNotFoundLabel: {
+    id: 'app.videoPreview.webcamNotFoundLabel',
+    description: 'Webcam not found label',
+  },
+});
+
+class VideoPreview extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      webcamDeviceId: null,
+      availableWebcams: null,
+      isStartSharingDisabled: false,
+    };
+
+    const {
+      closeModal,
+      startSharing,
+      changeWebcam,
+    } = props;
+
+    this.handleJoinVideo = this.handleJoinVideo.bind(this);
+    this.closeModal = closeModal;
+    this.startSharing = startSharing;
+    this.changeWebcam = changeWebcam;
+
+    this.deviceStream = null;
+  }
+
+  stopTracks() {
+    if (this.deviceStream) {
+      this.deviceStream.getTracks().forEach((track) => {
+        track.stop();
+      });
+    }
+  }
+
+  handleSelectWebcam(event) {
+    const webcamValue = event.target.value;
+    this.setState({ webcamDeviceId: webcamValue });
+    this.changeWebcam(webcamValue);
+    const constraints = {
+      video: {
+        deviceId: webcamValue ? { exact: webcamValue } : undefined,
+      },
+    };
+    this.stopTracks();
+    navigator.mediaDevices.getUserMedia(constraints).then((stream) => {
+      this.video.srcObject = stream;
+      this.deviceStream = stream;
+    }).catch((error) => {
+      console.log(error);
```

Comment:
The client side logger could be used to log this error message. [http://docs.bigbluebutton.org/html/html5-dev.html#client-logger](url)